tests/epics/areadetector/test_controllers.py

```diff
@@ -52,7 +52,7 @@ async def test_ad_controller(RE, ad: ADSimController):
 
 async def test_pilatus_controller(RE, pilatus: PilatusController):
     with patch("ophyd_async.core.signal.wait_for_value", return_value=None):
-        await pilatus.arm(trigger=DetectorTrigger.constant_gate)
+        await pilatus.arm(num=0, trigger=DetectorTrigger.constant_gate)
```

Comment:
Same as above; we should reserve `num=0` for flyscanning and use `num=1` here instead.

Reply:
In this pilatus case we are asserting that n=0 results in `999_999` as the value, are we not wanting to retain this test case?

Reply:
Done